megatron/training.py

```diff
@@ -785,14 +780,46 @@ def train_step(forward_step_func, data_iterator,
     return {}, skipped_iter, grad_norm, num_zeros_in_grad
 
 
+class interop_tool_logger:
+    def __init__(self, tb_writer=None, wandb_writer=None):
+        self.tb_writer = tb_writer
+        self.wandb_writer = wandb_writer
+
+    def is_enabled(self):
+        return self.tb_writer or self.wandb_writer
+
+    def add_scalar(self, key, scalar_value, step=None, tool_type='all'):
```

Comment:
How about making `tool_type`, perhaps with different naming, a list. This way clients can explicitly specify the tools they want updated. This will avoid negative tests such as `... and tool_type != ` which is quite brittle.

Reply:
Fix it. Thanks!